go-controller/pkg/factory/handler.go

```diff
@@ -103,8 +103,19 @@ func (i *informer) forEachQueuedHandler(f func(h *Handler)) {
 	i.RLock()
 	defer i.RUnlock()
 
-	for priority := 0; uint32(priority) <= minHandlerPriority; priority++ { // loop over priority higest to lowest
-		for _, handler := range i.handlers[uint32(priority)] {
+	for priority := 0; priority <= minHandlerPriority; priority++ { // loop over priority higest to lowest
+		for _, handler := range i.handlers[priority] {
+			f(handler)
+		}
+	}
+}
+
+func (i *informer) forEachQueuedHandlerReversed(f func(h *Handler)) {
+	i.RLock()
+	defer i.RUnlock()
+
+	for priority := minHandlerPriority; priority >= 0; priority-- { // loop over priority lowest to highest
```

Comment:
lgtm, main logic is here, makes sense that we tear down in the opposite order of creation dependency wise.